src/release.ss

```diff
@@ -0,0 +1,107 @@
+    show-console: #f stderr-redirection: #f]
+   read-all-as-string))
+
+;; Just run the process on the current console
+(def (process-run path . arguments)
+  (call-with-output-process
+   [path: path arguments: arguments
+    show-console: #f stderr-redirection: #f stdin-redirection: #f stdout-redirection: #f]
+   void))
+
+;; Compute next development version from the current release version
+(def (next-dev-version version)
+  (string-append version "-DEV"))
+
+;; Compute next release version from the current development version
+(def (next-release-version version)
+  (let* ((dash-position (string-index version #\-))
+         (bare-version (if dash-position (substring version 0 dash-position) version))
+         (dot-position (string-index-right bare-version #\.))
+         (prefix (substring bare-version 0 (+ dot-position 1)))
+         (last-component (substring bare-version (+ dot-position 1) (string-length bare-version)))
+         (number (string->number last-component)))
+    (format "~a~d" prefix (+ number 1))))
+
+;; Make a development version commit
+(def (make-dev-commit version)
+  (printf "Making commit for development version ~a~%" version)
+  ;; Remove the version file from git
+  (process-run "git" "rm" version-path)
+  ;; Add the version file back to .gitignore
+  (call-with-output-file [path: ignore-path append: #t]
+    (lambda (port) (fprintf port "~a\n" version-path)))
+  ;; Commit that
+  (process-run
+   "git" "commit"
+   (format "--message=Start development for version ~a" version)
+   version-path ignore-path)
+  ;; Tag the new commit with the desired version
+  (process-run "git" "tag" "--force" version))
+
+;; Make a release commit
+(def (make-release-commit version)
+  (printf "Making commit for release version ~a~%" version)
+  ;; Create the version file
+  (call-with-output-file [path: version-path truncate: #t]
+    (lambda (port) (fprintf port "(define (gerbil-version-string) \"~a\")\n" version)))
+  ;; Remove it from .gitignore
+  (def ignore-lines (remove version-path (read-file-lines ignore-path)))
+  (call-with-output-file [path: ignore-path truncate: #t]
+    (lambda (port) (for-each (lambda (line) (fprintf port "~a\n" line)) ignore-lines)))
+  ;; Add and commit those files
+  (process-run "git" "add" version-path ignore-path)
+  (process-run
+   "git" "commit"
+   (format "--message=Gerbil-~a" version)
+   version-path ignore-path)
+  ;; Tag the new commit with the desired version
+  (process-run "git" "tag" "--force" version))
+
+;; Remove newlines at the end of a string
+(def (string-trim-newline string)
+  (string-trim-right string (char-set #\return #\newline)))
+
+;; Main function, to be run from the command line
+(def (main (next-version #f))
+  (def previous-git-version
+    (string-trim-newline (process-output-string "git" "describe" "--tags")))
+  (def previous-is-release?
+    (not (string-index previous-git-version (char-set #\-))))
+  (if previous-is-release?
```

Comment:
I think I would like to have an (optional?) argument for the script that specifies whether this is a development or release tag when the version is specified.
There are cases where we may want to do a release on top of a release, eg: for an out-of-line hot patch release.
The automagic toggle behaviour is fine for the default when no arguments are specified.

So:
```
./release.ss rel v0.12
./release.ss dev v0.13-DEV

-- same when anchored on v0.12-DEV
./release.ss
```

The automagic toggle behaviour should be triggered only if no arguments are specified.